server/db/connect.js

```diff
@@ -0,0 +1,8 @@
+const mongoose = require("mongoose");
+
+const connectDB = async (url) => {
+  await mongoose.connect(url);
```

Comment:
Thank you for helping me see a mistake, which is for another reason.

So `mongoose.connect()` returns a promise that resolves when initial connection is made and rejects when it has failed. I was trying to log a success message 'Connected to MongoDB' upon a successful connection, which requires the _parent_ `connectDB` function to be async so the `console.log()` inside can await `mongoose.connect()` resolves--or else the log will take place immediately, before `mongoose.connect()` gets the chance to return the promise.

The mistake is that, without _returning_ `mongoose.connect()` 's promise in the `connectDB` function, any error will not be caught when `connectDB` is called inside the `start` function's try-catch blocks in `/app.js`. So the correct way of handling the mongoDB connection and starting the server, is to simply return `mongoose.connect()` in `connectDB` and move the console log of success message into the `start` function's `try` block, as a callback to the `connectDB` function, or immediately after `await` it and before starting the server; thanks to you raising the flag.

Reply:
Thank you for explaining what you were trying to achieve using the async/await function here. I do see that you would need the console.log to occur only after the connection promise is returned. 

There are other methods for error handling with mongoose.connect() - I have used this method which takes a callback function: https://mongoosejs.com/docs/connections.html#callback

```
mongoose.connect(process.env.DB_URL, {useNewUrlParser: true, useUnifiedTopology: true}, (err)=> {
    if(!err) {
        console.log("Successful connection with MongoDB server");
    } else {
        console.log("Error with MongoDB's connectivity");
    }
});
```

In case you are curious about the options I have used in the code snippet above: 
{useNewUrlParser: true}
https://mongoosejs.com/docs/5.x/docs/deprecations.html#the-usenewurlparser-option

{useUnifiedTopology: true}
https://mongoosejs.com/docs/5.x/docs/deprecations.html#useunifiedtopology

I like the callback function method because it allows error handling.

Reply:
Thanks for taking time again with this item. The options you reference are deprecated with Mongoose version ^6.0, and we're using Mongoose v.6.6.0, which comes with said options of course for `Mongoose.connect()`. See [here](https://mongoosejs.com/docs/migrating_to_6.html) for [No More Deprecation Warning Options](https://mongoosejs.com/docs/migrating_to_6.html#no-more-deprecation-warning-options). 

Absent further input of questionable matter with `server/db/connect`, your suggestions will be adopted in that the `server-sprint-2` branch, which will be proposed with changes from these comments and the inadequacy in the current GET all todo items call, will handle db connection in `/server/db/connect.js` as:
```
const connectDB = (url) => mongoose.connect(url);
```
and in `/server/app.js` as:
```
const start = async () => {
   try {
      await connectDB(process.env.MONGODB_URI);
      console.log("Connected to MongoDB");
      app.listen(port, () => console.log(`Server is listening on port ${port} ...`));
   } catch (err) {
      console.log(err);
   }
};
```
This way any errors from initiating db connection or starting the server will properly be handled in the order. The use of async-await or callbacks are entirely discretionary, and more voices in the community prefer the former for clarity.